### ISSUE TITLE
Saving a content block with specific text in the editor results in 404 error.

### ISSUE THREAD

**Issue by giulianoriccio** [magento/magento2#6582]
### Preconditions
1. Magento CE 2.1.1
2. PHP 5.6
### Steps to reproduce
1. Create or Edit a content block and input a specific content (see below)
2. Click Save
### Expected result
1. The page saves the block succesfully
### Actual result
1. The form post returns 404 and a 404 message appears on screen.

The bug is really weird.
If I input the following content, the block is saved:

``` html
<div class="fasciaverde banner banner-full">
    <div class="container">
        <div class="banner-inner-content">
            <p>Chat with us on Whatsapp. Find out more about Easy Chain</p>
            <span>39 312 123456789</span>
        </div>
    </div>
</div>
```

but if a "+" is added before 39 a 404 is returned:

``` html
<div class="fasciaverde banner banner-full">
    <div class="container">
        <div class="banner-inner-content">
            <p>Chat with us on Whatsapp. Find out more about Easy Chain</p>
            <span>+39 312 123456789</span>
        </div>
    </div>
</div>
```

however if the string in the paragraph is prefixed by a "p" the block is saved:

``` html
<div class="fasciaverde banner banner-full">
    <div class="container">
        <div class="banner-inner-content">
            <p>pChat with us on Whatsapp. Find out more about Easy Chain</p>
            <span>+39 312 123456789</span>
        </div>
    </div>
</div>
```

replacing the "p" tag with a "span", a 404 is returned again:

``` html
<div class="fasciaverde banner banner-full">
    <div class="container">
        <div class="banner-inner-content">
            <span>pChat with us on Whatsapp. Find out more about Easy Chain</span>
            <span>+39 312 123456789</span>
        </div>
    </div>
</div>

```

but replacing the prefixing "p" in the string with "span" the block is saved again:

``` html
<div class="fasciaverde banner banner-full">
    <div class="container">
        <div class="banner-inner-content">
            <span>spanChat with us on Whatsapp. Find out more about Easy Chain</span>
            <span>+39 312 123456789</span>
        </div>
    </div>
</div>
```

**Comment by stamster** [comment]
I believe this issue has something to do with regular expressions which obviously handle input on Content blocks / pages etc. 
Both `+` and `p` do have a meaning in REGEXP.

Additionally, recently there's been a few updates for libpcre3(-dev), where there are reports of weird behaviour from other projects (non-Magento), so check also that which package version do you run, in this case try with a pre-June 2016 versions (even though sounds really weird and would impact much larger scale audience).

**Comment by giulianoriccio** [comment]
I thought the same thing. The `p` is relative to the wrapping tag, infact if I use a span the prefixing string must be `span` to not throw an error.

I'll try installing it on my dev machine and see if the same thing happens.

**Comment by giulianoriccio** [comment]
This is the PCRE version I got

```
$ pcretest -C
PCRE version 7.8 2008-09-05
```

**Comment by anton-huz** [comment]
Hi, Giuliano,
I've tried to reproduce your issue, and I have not faced the _"Actual result"_.
- Are you sure that you doesn't have an extension to your browser (like Skype add-on, that tries to parse phone numbers from the page)? Could you try same steps in _Private mode_?
- Could you observe any reports/logs from `var/` directory of Magento installation? It would be handy  to attache them here.

**Comment by giulianoriccio** [comment]
Hi @guz-anton,
after further investigation, I have found out that the issue is not with Magento but probably with Nexcess.

I'm sorry for wasting your time.

I replaced the index.php with custom content but it had no effect. After stripping down the htaccess, the error being shown became a 403 (a custom 404 page was being shown for the 403 error).

Since the SSH access is limited and I've no access to the server logs, I'll contact Nexcess to try to find out why the server is returning a 403 when that specific content is being sent.

Thank you for your time.

EDIT: looks like the culprit is mod_security
